docs/tutorials/migrating-your-ga4-data-to-snowplow/index.md

```diff
@@ -0,0 +1,227 @@
+`app_id`, for example, must be changed, substituting the placeholder values for the correct values for your GA4 data.\
+The SQL that you input must conform to strict requirements:
+- The SQL written for the mapping will be placed directly into the code, so it must be valid as a standalone single-line statement:
+```sql
+...
+<your input SQL> AS snowplow_field
+...
+```
+| Snowplow_field | mapping_sql |
+| --------------- | ----------- |
+| page_urlscheme  | regexp_extract(page_url, r'(.*?):') |
+| page_urlhost    | {{dbt_utils.get_url_host(field='page_url')}} |
+| geo_zipcode     | NULL        |
+| app_id          | CASE WHEN stream_id = '4271243942' THEN 'Snowplow' ELSE NULL END |
+
+For example, at compile time, the above will look like:
+```SQL
+...
+regexp_extract(page_url, r'(.*?):') AS page_urlscheme,
+{{dbt_utils.get_url_host(field='page_url')}} AS page_urlhost,
+"NULL" AS geo_zipcode,
+CASE WHEN stream_id = '4271243942' THEN 'Snowplow' ELSE NULL END AS app_id,
+...
+```
+- Some Snowplow fields have been pre-extracted from the GA4 data to make the SQL more readable, and to make using these fields in further calculations easier. These are:
+    - `domain_sessionidx`
+    - `page_url`
+    - `page_title`
+    - `page_referrer`
+    - `mkt_medium`
+    - `mkt_source`
+    - `mkt_term`
+    - `mkt_content`
+    - `mkt_campaign`
+    - `domain_sessionid`
+    
+As such, when writing SQL in the mapping, you can use these fields directly, e.g.:
+| Snowplow_field | mapping_sql |
+| --------------- | ----------- |
+| page_urlquery   | regexp_extract(page_url, r'(?:.*)\\?(.*)')|
+- If you wish to use a parameter from `event_params`, we have a UDF that will extract the parameter from the event_params column. The UDF is called `ga4_migrator_parse_params`. To use it, the SQL must be written in the following format:
+```sql
+JSON_VALUE(ga4_migrator_parse_params(event_params), '$.parameter_to_extract')
+```
+E.g.:
+```sql
+JSON_VALUE(ga4_migrator_parse_params(event_params), '$.page_title')
+```
+- When creating or updating any of the mapping fields and using a dbt package, it must be written in the following format, including the curly braces and the package reference: `{{dbt_package.function('column_name')}}`, e.g. `{{dbt_utils.get_url_host('page_referrer')}}`.
+
+- In the mapping table, there are two fields that should not be altered:\
+`geo_country_temp` and `geo_region_temp`. These are required for the `geo_country` and `geo_region` fields to be populated correctly.
+
+### Step 4 - Transform data
+
+Click the `Run transformation` button to run the migration. Behind the scenes, this sends a request to the API, which will run dbt. dbt will create the UDF that extracts the event_params, seed two tables that are used for geographical mapping, and create the final table that contains the migrated data.\
```

Comment:
"sends a request to the API" the importer tool encapsulates the API, from the user perspective I don't think this needs to be pointed out. Probably start with DBT.